app/helpers/application_helper.rb

```diff
@@ -18,6 +18,7 @@ def escape_json(hash)
     # using json_escape to prevent XSS vulnerability
     str = json_escape(hash.to_json) unless hash.class == 'String'
     str = str.gsub!("\\", "\\\\\\") if str.include? "\\"
+    str = str.gsub("'", "\\\\'")
```

Comment:
it appears this isn't the first time we've ran into it,  i would git blame line 20 and see who added that :)

also seems like this truly belongs inside json_escape;   may be worth creating a security issue

Reply:
Yes, line 20 was added during the big frontend migration. We couldn't figure out the right solution at the time and still haven't...